### PR TITLE
Push prow images with make rules to staging registry

### PR DESCRIPTION
**Pull request by chaodaiG**
/hold
Depend on #25119 

Part of: #23832 

/cc @BenTheElder

### CODE REVIEW

File: config/jobs/kubernetes/test-infra/test-infra-trusted.yaml [hunk 1]
```diff
@@ -221,6 +221,39 @@ postsubmits:
     rerun_auth_config:
       github_users:
       - alvaroaleman
+  - name: post-test-infra-push-prow-staging
+    cluster: test-infra-trusted
+    # Runs on more than just the Prow dir to include some additional images that we publish to gcr.io/k8s-prow.
+    run_if_changed: '^(WORKSPACE|load.bzl|repos.bzl|containers.bzl|prow|ghproxy|label_sync/.+\.go|robots/commenter|robots/pr-creator|robots/issue-creator|testgrid/cmd|gcsweb)'
```
Comment (cjwagner) [inline config/jobs/kubernetes/test-infra/test-infra-trusted.yaml:227]: This is no longer accurate. What is the plan for all these images?

File: config/jobs/kubernetes/test-infra/test-infra-trusted.yaml [hunk 2]
```diff
@@ -221,6 +221,39 @@ postsubmits:
...
+    decorate: true
+    branches:
+    - ^master$
+    max_concurrency: 1
+    spec:
+      serviceAccountName: pusher
+      containers:
+      - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20220124-681f3531ec-test-infra
+        command:
+        - runner.sh
```
Comment (cjwagner) [inline config/jobs/kubernetes/test-infra/test-infra-trusted.yaml:237]: What is the motivation for using the kubekins-e2e image and `runner.sh`?

Comment (chaodaiG) [reply config/jobs/kubernetes/test-infra/test-infra-trusted.yaml:237]: still need dind for building typescript

Comment (chaodaiG) [reply config/jobs/kubernetes/test-infra/test-infra-trusted.yaml:240]: now that you reminded me, updated the job to use dind

File: config/jobs/kubernetes/test-infra/test-infra-trusted.yaml (hunk 1 above)
Comment (chaodaiG) [reply config/jobs/kubernetes/test-infra/test-infra-trusted.yaml]: updated to be more accurate

File: config/jobs/kubernetes/test-infra/test-infra-trusted.yaml [hunk 3]
```diff
@@ -221,6 +221,45 @@ postsubmits:
     rerun_auth_config:
       github_users:
       - alvaroaleman
+  - name: post-test-infra-push-prow-staging
+    cluster: test-infra-trusted
+    # Runs on more than just the Prow dir to include some additional images that we publish to gcr.io/k8s-prow.
+    run_if_changed: '^(hack/make-rules|prow|ghproxy|label_sync/.+\.go|robots/commenter|robots/pr-creator|robots/issue-creator|testgrid/cmd|gcsweb)'
```
Comment (cjwagner) [inline config/jobs/kubernetes/test-infra/test-infra-trusted.yaml:227]: We're still including a bunch of directories for images that are not pushed by this job, right?
Are you planning to continue publishing these images with the core Prow images?

**Review by cjwagner — APPROVED** [review]
I still have a question about the run_if_changed expression and the images it references.
/hold

**Pull request merged**